### PR TITLE
Bug fix: calculation of dry months on rainfall quality control

### PR DESCRIPTION
**Pull request by shadrackkibet**
Fixes part of #6115 and #5017. @rdstern this fixes the problem with the calculation of dry months on rainfall quality control.

Notes

- Added method that checks if nudUpDown is empty.
- Code refactor: sorted out parameter positions, test Ok fix and change to how `calculated_from` parameter is added into calculations.
- Bugfix: Ensuring that a monthly filter is not added when there are no months to omit.
- Changing the threshold label for dry months option.

### CODE REVIEW

**Comment by shadrackkibet** [comment]
@africanmathsinitiative/developers this is ready for review.

**Comment by lloyddewit** [comment]
@Ivanluv please could you review? thanks

**Comment by Ivanluv** [comment]
@lloyddewit  am looking at this now

**Comment by Ivanluv** [comment]
@lloyddewit  the code looks okay from my end

**Comment by lloyddewit** [comment]
@rdstern please could you test? thanks

**Review by rdstern — DISMISSED** [review]
Excellent that this seems resolved.

File: instat/dlgClimaticCheckDataRain.vb [hunk 1]
```diff
@@ -630,25 +625,13 @@ Public Class dlgClimaticCheckDataRain
...
-            ucrBase.OKEnabled(True)
-        Else
-            ucrBase.OKEnabled(False)
-        End If
-        If ucrChkOutlier.Checked AndAlso ucrChkOmitZero.Checked AndAlso ucrInputThresholdValue.IsEmpty Then
+        If ucrReceiverElement.IsEmpty OrElse ((ucrChkLarge.Checked AndAlso ucrNudLarge.IsEmpty) OrElse (ucrChkSame.Checked AndAlso (ucrNudSame.IsEmpty OrElse ucrInputSameValue.IsEmpty)) OrElse (ucrChkWetDays.Checked AndAlso ucrNudWetDays.IsEmpty) OrElse (ucrChkOutlier.Checked AndAlso (ucrNudSkewnessWeight.IsEmpty OrElse ucrNudCoeff.IsEmpty)) OrElse (ucrChkDryMonth.Checked AndAlso (ucrInputThreshold.IsEmpty OrElse ucrReceiverMonth.IsEmpty))) OrElse ucrChkOutlier.Checked AndAlso (ucrChkOmitZero.Checked AndAlso ucrInputThresholdValue.IsEmpty) Then
```
Comment (lloyddewit) [inline instat/dlgClimaticCheckDataRain.vb:628]: More readable if lines don't exceed 100 columns (but this issue is not caused by your change)
```suggestion
        If ucrReceiverElement.IsEmpty OrElse 
                ((ucrChkLarge.Checked AndAlso ucrNudLarge.IsEmpty) OrElse 
                (ucrChkSame.Checked AndAlso (ucrNudSame.IsEmpty OrElse ucrInputSameValue.IsEmpty)) OrElse 
                (ucrChkWetDays.Checked AndAlso ucrNudWetDays.IsEmpty) OrElse 
                (ucrChkOutlier.Checked AndAlso (ucrNudSkewnessWeight.IsEmpty OrElse ucrNudCoeff.IsEmpty)) OrElse
                (ucrChkDryMonth.Checked AndAlso (ucrInputThreshold.IsEmpty OrElse ucrReceiverMonth.IsEmpty))) OrElse
                ucrChkOutlier.Checked AndAlso (ucrChkOmitZero.Checked AndAlso ucrInputThresholdValue.IsEmpty) Then
```

File: instat/ucrNud.vb [hunk 2]
```diff
@@ -153,6 +153,10 @@ Public Class ucrNud
         End If
     End Function
 
+    Public Function IsEmpty() As Boolean
+        Return nudUpDown.Text = ""
```
Comment (lloyddewit) [inline instat/ucrNud.vb:157]: is this safer?
```suggestion
        Return string.IsNullOrEmpty(nudUpDown.Text)
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@shadrack looks good, just a couple of small comments

**Comment by shadrackkibet** [comment]
No functionality change. Happy with these changes.

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern Small changes since your last approval (functionality not affected). Please could you approve again? thanks

**Comment by shadrackkibet** [comment]
@rdstern ?

**Review by rdstern — APPROVED** [review]

**Pull request merged**